### ISSUE TITLE
Use index when there are multiple label options 

### ISSUE THREAD

**Issue by grillorafael** [neo4j/neo4j#12619]
## Feature request template

**Is your feature request related to a problem? Please describe.**
I often have relationships that can happen at different node labels but I don't always know beforehand which node label it is. 

When I run this:
```cypher
match (n {id: '656ba4d4-09a2-42e6-a5ae-4dbf1603d5df'})
where n: GeneratingUnit  or n: Continent 
return n;
```
neo4j does not use the id indexes that exist for GeneratingUnit or Continent but instead does a scan by label.

If I run a union all query it does behave how I wish it would:
![image](https://user-images.githubusercontent.com/1812267/98273567-53fd7700-1f8a-11eb-8385-5361d7fc518f.png)

**Describe the solution you'd like**
I would like neo4j to try to find a index to use when there are multiple labels as a OR statement

**Describe alternatives you've considered**
Using unionall triggers the behavior I want but it is not clean

**Comment by mnd999** [comment]
Thanks for the feature request, it certainly looks like something that could be useful. I've passed it to our product team who will look into it.

**Comment by hedengran** [comment]
Hi, sorry for not getting back sooner. This feature has been implemented and merged in version 4.3.

**Comment by grillorafael** [comment]
Thanks @hedengran
